src/orion/core/worker/warm_start/multi_task_wrapper.py

```diff
@@ -0,0 +1,245 @@
+        # experiments in the KB, in case some get added in the future?
+        # TODO: Should the number of tasks here only count the experiments where the spaces are
+        # compatible? (Currently only counts total number of experiments in the KB)
+        # TODO: Could we use a Catagorical over the experiment IDS instead of a Categorical with
+        # ints? Would that help in some way?
+        max_task_id = knowledge_base.n_stored_experiments + 1
+        task_label_dimension = Categorical(
+            "task_id",
+            list(range(0, max_task_id)),
+            default_value=0,
+        )
+        new_space = copy.deepcopy(space)
+        if "task_id" in new_space:
+            raise RuntimeError(f"Space already has a task_id dimension: {new_space}")
+        new_space.register(task_label_dimension)
+        return new_space
+
+    def _is_compatible(self, trial_from_other_experiment: Trial) -> bool:
+        """Used to check if a trial is compatible with the current experiment and can be reused for warm-starting.
+        The trial will have a different task ID than the current task.
+
+        NOTE: this assumes that the trial from the other experiment comes from the knowledge base,
+        and that is therefore doesn't already have a task ID.
+        """
+        # TODO: Do we need to do something smarter here?
+        return trial_from_other_experiment in self.space
+
+    def warm_start(self, warm_start_trials: dict[ExperimentInfo, list[Trial]]) -> None:
+        """Use the given trials to warm-start the algorithm.
+
+        These experiments and their trials were fetched from some knowledge base, and
+        are believed to be somewhat similar to the current on-going experiment.
+
+        It is the responsibility of the Algorithm to implement this method in order to
+        take advantage of these points.
+
+        Parameters
+        ----------
+        warm_start_trials : Dict[ExperimentInfo, List[Trial]]
+            Dictionary mapping from ExperimentInfo objects (containing the experiment config) to
+            the list of Trials associated with that experiment.
+        """
+        # Perform warm-starting using only the supported trials.
+        log.info(
+            "Will warm-start using contextual information, since the algo isn't warm-starteable."
+        )
+
+        # Dict mapping from task ID to list of compatible trials.
+        compatible_trials: dict[int, list[Trial]] = defaultdict(list)
+        task_ids_to_experiments: dict[int, ExperimentInfo] = {}
+
+        for i, (experiment_info, trials) in enumerate(warm_start_trials.items()):
+            # Start the task ids at 1, so the current experiment has task id 0.
+            task_id = i + 1
+            for trial in trials:
+                # Drop the point if it doesn't fit inside the current space.
+                # TODO: Do we want to 'translate' the point in this case?
```

Comment:
Maybe this kind of feature should be implemented on the side of the KB so that the user can configure the KB instead of the algo wrapper for this. Or maybe something in-between the KB and the algo wrapper.